### PR TITLE
refactor: Apply new Angular control flow

### PR DESCRIPTION
**Pull request by dominik003**
This PR applies the new Angular control flow introduced with Angular 17 to several components. This includes some simplifications that can now be applied more easily. In addition, some css classes have been converted to tailwind and unused css files have been removed. Last but not least, some line breaks and indentations in the components have been adjusted to improve readability.

### CODE REVIEW

File: frontend/src/app/projects/models/backup-settings/create-backup/create-backup.component.html [hunk 1]
```diff
@@ -5,69 +5,70 @@
...
+              *ngFor="let t4cModel of t4cModelService.t4cModels$ | async"
+              [value]="t4cModel.id"
+            >
+              <div mat-line>TeamForCapella model '{{ t4cModel.name }}'</div>
+              <div mat-line>
+                repository '{{ t4cModel.repository.name }}', instance '{{
+                  t4cModel.repository.instance.name
+                }}'
+              </div>
+            </mat-list-option>
```
[the remark this answers is not in the document]
Comment (romeonicholas) [reply frontend/src/app/projects/models/backup-settings/create-backup/create-backup.component.html:24]: I haven't actually tested this, just saw a few ngfors that could probably be easily swapped to the new format as well

**Review by romeonicholas — COMMENTED** [review]
So the changes I'm seeing make sense, though there are a number of logic changes I haven't been able to personally test. I'm kind of wondering what the criteria for updating is, since you left some of the old logic controls in components you updated. Is this intended to be ongoing?

**Comment by dominik003** [comment]
> So the changes I'm seeing make sense, though there are a number of logic changes I haven't been able to personally test. I'm kind of wondering what the criteria for updating is, since you left some of the old logic controls in components you updated. Is this intended to be ongoing?

So there were a few cases that I wasn't quite sure how we should handle. If I remember correctly, they were:

- using `*ngIf` for `mat-error` and for certain, smaller `span` elements, as I wasn't quite sure if we want to use the new control flow there or if the normal inline approach is better
- using `*ngFor` for `mat-options` as I wasn't sure and didn't have time to check if using the new control flow works as expected

I think I'll use your comments to think again about whether it makes sense to apply the new control flow to all conditions and loops, or whether we should perhaps exclude some scenarios in the next planning to keep it simple.

And finally, this PR only serves as a starting point for the new control flow. This is mainly because this PR already contains a large number of changes and I didn't want to make it too big. Also, applying these changes takes some time, as the new control flow has a lot of room for major simplifications, especially since we can now use `else if` and `else` properly.

**Comment by MoritzWeber0** [comment]
I really appreciate the effort you spent on the changes, but some components are really hard to test. Testing all of the modified changes will require a significant effort. 

This week we've introduced storybook in the Capella Collaboration Manager. Storybook offers a nice UI and with the Chromatic integration stories will automatically scanned for visual changes. Due to mocking, the effort to implement stories low. 

Therefore, we should have a story defined for each component that is refactored to avoid some unintentional changes.

**Comment by MoritzWeber0** [comment]
Due to the amount of conflicts, I'll close the PR. Instead, we'll iteratively switch to the new format, together with the introduction of stories of the the affected components.

**Comment by MoritzWeber0** [comment]
Due to the amount of conflicts, I'll close the PR. Instead, we'll iteratively switch to the new format, together with the introduction of stories for the the affected components.

**Pull request closed without merging**